slidechain/export.go

```diff
@@ -248,12 +321,13 @@ func SubmitPreExportTx(hclient horizon.ClientInterface, kp *keypair.Full, custod
 	return temp.Address(), seqnum, nil
 }
 
-// BuildExportTx builds a txvm retirement tx for an asset issued
-// onto slidechain. It will retire `amount` of the asset, and the
-// remaining input will be output back to the original account.
-func BuildExportTx(ctx context.Context, asset xdr.Asset, amount, inputAmt int64, temp string, anchor []byte, prv ed25519.PrivateKey, seqnum xdr.SequenceNumber) (*bc.Tx, error) {
-	if inputAmt < amount {
-		return nil, fmt.Errorf("cannot have input amount %d less than export amount %d", inputAmt, amount)
+// BuildExportTx locks money to be retired in a TxVM smart contract.
+// If the peg-out transaction succeeds, it retires `exportAmt` of this asset,
+// and `inputAmt` - `exportAmt` will be output back to the original account.
+// Else, it returns `exportAmt` to the exporter address.
```

Comment:
Nit: the wording needs a tweak here to make it clearer that you're describing what the smart contract does (when invoked separately later), not what this function does.